salt/states/pem.py

```diff
@@ -0,0 +1,173 @@
+"""
+Provide easier operations on certificate files
+=====================================================================
+
+Main purpose for this is to provide easier and human readable info
+regarding changes if files contains certificates.
+
+Requirement is `cryptography` python package.
+
+.. code-block:: yaml
+
+/etc/ssl/postfix/mydomain/mydomain.pem:
+  pem.managed:
+    - source: salt://files/etc/ssl/mydomain.pem
+    - user: root
+    - group: postfix
+    - dir_mode: 711
+    - makedirs: True
+    - template: jinja
+"""
+
+import logging
+
+import salt.utils.files
+
+try:
+    from cryptography import x509
+
+    HAS_CRYPTOGRAPHY = True
+except ImportError:
+    HAS_CRYPTOGRAPHY = False
+
+__virtualname__ = "pem"
+
+
+def __virtual__():
+    if not HAS_CRYPTOGRAPHY:
+        return (False, "Could not load cryptography")
+    return __virtualname__
+
+
+log = logging.getLogger(__name__)
+
+
+def managed(
+    name,
+    source=None,
+    source_hash=None,
+    source_hash_name=None,
+    user=None,
+    group=None,
+    mode=None,
+    skip_verify=None,
+    defaults=None,
+    attrs=None,
+    context=None,
+    saltenv="base",
+    skip_conditions=False,
+    **kwargs,
+):
+    """
+    State manage certificates as files and provide minimal overview of
+    expiration and common name in comment.
```

Comment:
Manage certificates as files and provide certificate expiration and common name in the comment.